File: src/client/src/app/modules/admin/payroll/models/payrollRequest.ts
import { PayPeriod } from "src/app/core/enums/PayPeriod";

export interface PayrollRequest {
    id: string;
    createaAt: string;
    updatedAt: string;
    organizationId: string;
    branchId: string;
    startDate: Date;
    endDate: Date;
    month: number;
    payPeriod: PayPeriod;
    salaryCalculationFormula: any;
    ignoreAttendance: boolean;
    ignoreDeductionForAbsents: boolean;
    ignoreDeductionForLateComer: boolean;
    ignoreOvertime: boolean;
    monthName: string;
    payPeriodName: string;
    status: string;
    startedAt?: Date;
    endedAt?: Date;
    run: boolean;
}

File: src/server/Modules/Accounting/Modules.Accounting.Core/Entities/PayrollRequest.cs
using FluentPOS.Shared.Core.Domain;
using FluentPOS.Shared.DTOs.Enums;
using System;

namespace FluentPOS.Modules.Accounting.Core.Entities
{
    public class PayrollRequest : BaseEntity
    {

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public int Month { get; set; }

        public PayPeriod PayPeriod { get; set; }

        public SalaryCalculationFormula SalaryCalculationFormula { get; set; }

        public bool IgnoreAttendance { get; set; }

        public bool IgnoreDeductionForAbsents { get; set; }

        public bool IgnoreDeductionForLateComer { get; set; }

        public bool IgnoreOvertime { get; set; }

        public string Status { get; set; }

        public DateTime? StartedAt { get; set; }

        public DateTime? EndedAt { get; set; }

        public string Logs { get; set; }

        public string Message { get; set; }
    }
}